### PR TITLE
[KOGITO-2650] Setup retry HTTP calls for tests

### PR DESCRIPTION
**Pull request by radtriste**
https://issues.redhat.com/browse/KOGITO-2650

Mainly to avoid Openshift and network stabilities.

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [ ] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

File: test/framework/http.go [hunk 1]
```diff
@@ -72,7 +74,27 @@ func WaitForSuccessfulHTTPRequest(namespace string, requestInfo HTTPRequestInfo,
...
-	return ExecuteHTTPRequestC(&http.Client{}, namespace, requestInfo)
+	// Setup a retry in case the first time it did not work
+	retry := 0
+	var resp *http.Response
+	var err error
+	for retry < config.GetHTTPRetryNumber() {
+		resp, err := ExecuteHTTPRequestC(&http.Client{}, namespace, requestInfo)
+		if err == nil && checkHTTPResponseSuccessful(namespace, resp) {
+			return resp, err
+		} else if resp.StatusCode == 500 {
```
Comment (Sgitario) [inline test/framework/http.go:85]: Looking in ExecuteHTTPRequestC, it seems that the response could return null in the response with an error happened.   (Scenario when err is not null and resp is null). So I guess, we should either do the else if err != nil first, or check whether resp is not nil here.

File: Makefile [hunk 2]
```diff
@@ -180,6 +181,7 @@ run-tests:
 		--container_engine ${container_engine} \
 		--domain_suffix ${domain_suffix} \
 		--image_cache_mode ${image_cache_mode} \
+		--http_retry_nb ${http_retry_nb} \
```
Comment (Sgitario) [inline Makefile:184]: I guess this should be covered in the bats tests too.

File: test/framework/http.go (hunk 1 above)
Comment (radtriste) [reply test/framework/http.go:85]: good catch. will update

File: test/framework/http.go [hunk 3]
```diff
@@ -72,7 +74,30 @@ func WaitForSuccessfulHTTPRequest(namespace string, requestInfo HTTPRequestInfo,
...
+		if err == nil && checkHTTPResponseSuccessful(namespace, resp) {
+			return resp, err
+		} else if err != nil {
+			GetLogger(namespace).Warnf("Error while making http call: %v", err)
+		} else if resp != nil {
+			if resp.StatusCode == 500 {
+				// In case of 500, it is server error and we don't need to call again
+				return resp, err
+			}
+			GetLogger(namespace).Warnf("Http call was not successful. Response code: %d", resp.StatusCode)
```
Comment (MarianMacik) [inline test/framework/http.go:92]: Maybe we could first log the message and then in case it is 500, return. Now it won't log the code if it is 500.

Comment (radtriste) [reply test/framework/http.go]: updated

**Pull request merged**